### ISSUE TITLE
XNote constantly causes Thunderbird to crash on Mac

### ISSUE THREAD

**Issue by zevrix** [xnotepp/xnote#54]
XNote constantly causes Thunderbird crashes on Mac.
XNote 3.1.3
macOS 10.15, 11.x
Thunderbird 78.7.0

By now I'm confident it's XNote: I especially ran Thunderbird with no add-ons for a week - no crashes at all. Re-enabled XNote (which is my only plug-in): crashes returned right away.

Happens at least once a day, sometimes multiple times.

The following actions can trigger the crashes: Save email draft, invoke Reply command, Send email, new email arrived (even if Thunderbird in the background).

I attached a crash report below.

I want to mention that I love XNote, use it for years and find it indispensable. I sent a donation too.

All the problems started with the recent drastic overhaul of TB (was it version 78?), which at first didn't support XNote at all.

Crash report is below, please let me know if you need more info. Should I report this bug on Bugzilla too - or is it XNote issue?

AdapterDeviceID: 0x67df
AdapterVendorID: 0x1002
Add-ons: xnote%40froihofer.net:3.1.3,%7B0D2172E4-C5AE-465A-B80D-53A840275B5E%7D:2.0.3,wetransfer%40extensions.thunderbird.net:2.1.0,default-theme%40mozilla.org:1.0,google%40search.mozilla.org:1.0,amazon%40search.mozilla.org:1.1,bing%40search.mozilla.org:1.0,ddg%40search.mozilla.org:1.0,wikipedia%40search.mozilla.org:1.0
BuildID: 20210126033722
ContentSandboxCapable: 1
ContentSandboxLevel: 1
CrashTime: 1612287366
EventLoopNestingLevel: 1
FramePoisonBase: 9223372036600930304
FramePoisonSize: 4096
InstallTime: 1611897783
Notes: FP(D00-L1000-W00000000-T000) WR? WR- OMTP? OMTP+4 GL Layers? GL Context? GL Context+ GL Layers+ 
ProductID: {3550f703-e582-4d05-9a08-453d09bdfdc6}
ProductName: Thunderbird
ReleaseChannel: release
SafeMode: 0
SecondsSinceLastCrash: 35815
StartupCrash: 0
StartupTime: 1612251554
TelemetryEnvironment: {"build":{"applicationId":"{3550f703-e582-4d05-9a08-453d09bdfdc6}","applicationName":"Thunderbird","architecture":"x86-64","buildId":"20210126033722","version":"78.7.0","vendor":null,"displayVersion":"78.7.0","platformVersion":"78.7.0","xpcomAbi":"x86_64-gcc3","updaterAvailable":true},"partner":{"distributionId":null,"distributionVersion":null,"partnerId":null,"distributor":null,"distributorChannel":null,"partnerNames":[]},"system":{"memoryMB":40960,"virtualMaxMB":null,"cpu":{"extensions":["hasMMX","hasSSE","hasSSE2","hasSSE3","hasSSSE3","hasSSE4_1","hasSSE4_2","hasAVX","hasAVX2","hasAES"]},"os":{"name":"Darwin","version":"20.1.0","locale":"en-US"},"hdd":{"profile":{"model":null,"revision":null,"type":null},"binary":{"model":null,"revision":null,"type":null},"system":{"model":null,"revision":null,"type":null}},"gfx":{"D2DEnabled":null,"DWriteEnabled":null,"ContentBackend":"Skia","Headless":false,"adapters":[{"description":null,"vendorID":"0x1002","deviceID":"0x67df","subsysID":null,"RAM":0,"driver":null,"driverVendor":null,"driverVersion":null,"driverDate":null,"GPUActive":true}],"monitors":[{"screenWidth":2560,"screenHeight":1440,"scale":2}],"features":{"compositor":"none","hwCompositing":{"status":"available"},"gpuProcess":{"status":"unused"},"wrQualified":{"status":"blacklisted:FEATURE_UNQUALIFIED_WEBRENDER_MAC"},"webrender":{"status":"opt-in"},"wrCompositor":{"status":"disabled"}}},"appleModelId":"iMac18,3"},"settings":{"blocklistEnabled":true,"e10sEnabled":false,"e10sMultiProcesses":8,"telemetryEnabled":false,"locale":"en-US","intl":{},"update":{"channel":"release","enabled":true},"userPrefs":{"browser.cache.disk.capacity":1048576,"browser.search.region":"CA","fission.autostart":false,"general.smoothScroll":false},"sandbox":{"effectiveContentProcessLevel":1},"addonCompatibilityCheckEnabled":true,"isDefaultBrowser":null},"profile":{}}
ThreadIdNameMapping: 5:"Gecko_IOThread",8:"Socket Thread",9:"Permission",10:"JS Watchdog",12:"JS Helper",13:"JS Helper",15:"JS Helper",11:"JS Helper",17:"JS Helper",16:"JS Helper",14:"JS Helper",18:"JS Helper",20:"Timer",21:"Cache2 I/O",22:"Cookie",24:"Worker Launcher",25:"Compositor",26:"ImgDecoder #1",27:"ImageIO",28:"IPDL Background",29:"DataStorage",30:"cert_storage",31:"DOM Worker",34:"mozStorage #1",37:"StyleThread#2",35:"StyleThread#0",36:"StyleThread#1",38:"StyleThread#3",39:"StyleThread#4",40:"StyleThread#5",42:"ImageBridgeChld",45:"TRR Background",46:"HTML5 Parser",47:"localStorage DB",48:"QuotaManager IO",53:"ImgDecoder #2",54:"mozStorage #4",56:"IMAP",57:"IMAP",58:"IMAP",59:"IMAP",60:"IMAP",62:"IMAP",63:"IMAP",64:"IMAP",65:"IMAP",66:"IMAP",71:"mozStorage #5",73:"mozStorage #6",78:"mozStorage #7",79:"mozStorage #8",78:"URL Classifier",71:"ImgDecoder #3",72:"ImgDecoder #4",72:"mozStorage #9",72:"DNS Resolver #32",67:"DNS Resolver #40",66:"IMAP",67:"DOM Worker",68:"IMAP",72:"StreamTrans #1853",69:"StreamTrans #1856",
Throttleable: 1
UptimeTS: 35812.23562711
Version: 78.7.0
useragent_locale: en-US

This report also contains technical information about the state of the application when it crashed.

**Comment by opto** [comment]
unfortunately, I do not have access to a Mac so I cannot test.
Some ideas/questions: I read it crashes on standard TB actions (like send email). Do these actions happen on emails with xnotes, or on unrelated emails? (obviously not possible for new email arrived, unless it is in a conversation with an email with xnotes.

Klaus

**Comment by zevrix** [comment]
Thanks Klaus,

Nope there's no obvious relation to Xnote - the crashes happen on random emails many of them with no Xnote assigned.

I'll then submit the bug to Bugzilla too - maybe they have answers.

I wish they never made all the drastic changes in TB 78 that only resulted in incompatibilities and other numerous problems.

**Comment by wsmwk** [comment]
I agree, there is not a strong correlation to xnote. For example the user of this crash does not have xnote installed https://crash-stats.mozilla.org/report/index/e636f285-b898-4bd2-ace2-cb7c90210227#tab-extensions

**Comment by zevrix** [comment]
> I agree, there is not a strong correlation to xnote. 

if it's not XNote that's causing the crashes then it's even better. all i know is that i especially ran TB with add-ons disabled and it never crashed. XNote was the only one i used as far as i know.

**Comment by opto** [comment]
the no crashing and the not crashing TB -  are they the same version

**Comment by zevrix** [comment]
> the no`t crashing and the crashing TB - are they the same (sub-) version

yes, any 78.x

**Comment by opto** [comment]
fishing in the dark ... just saw a crash problem related to master password: https://thunderbird.topicbox.com/groups/addons/Tf4607f56bf406d16/addon-problems-with-primary-password-enabled

maybe something is similiar/related?

**Comment by opto** [comment]
please try new version (later tonight) on github releases for xnote++

**Comment by Chealer** [comment]
@zevrix: Does this persist with 4.3.38?
